Product.cs

```diff
@@ -14,12 +14,13 @@ public class Product
         public static decimal BasicPrice { get; set; }
 
         decimal tax;
-        decimal discount;
+        decimal universalDiscount;
+        decimal upcDiscount;
```

Comment:
you do not have to put these properties here too, but as your program depends on the constructor to calculate (which is something wrong) you have to do so, 
instead, we can separate the data (product with only it's related info: price, UPC) from the calculations and its data (tax, discount)
as the tax and discount are not properties of a real-world product.
let's also discuss that